Sources/PubNub/EventEngine/Presence/Helpers/PresenceHeartbeatRequest.swift

```diff
@@ -0,0 +1,148 @@
+//
+//  PresenceHeartbeatRequest.swift
+//
+//  Copyright (c) PubNub Inc.
+//  All rights reserved.
+//
+//  This source code is licensed under the license found in the
+//  LICENSE file in the root directory of this source tree.
+//
+
+import Foundation
+
+// MARK: - PresenceStateContainer
+
+class PresenceStateContainer {
+  private var channelStates: Atomic<[String: [String: JSONCodableScalar]]> = Atomic([:])
+  private var channelGroupStates: Atomic<[String: [String: JSONCodableScalar]]> = Atomic([:])
+  
+  static var shared: PresenceStateContainer = PresenceStateContainer()
+  private init() {}
+  
+  func registerState(_ state: [String: JSONCodableScalar], forChannels channels: [String]) {
+    channelStates.lockedWrite { channelStates in
+      channels.forEach {
+        channelStates[$0] = state
+      }
+    }
+  }
+  
+  func registerState(_ state: [String: JSONCodableScalar], forChannelGroups groups: [String]) {
```

Comment:
That's true. I provided parameter for channel groups in case supporting it by the server anytime in the future. I can easily remove it since it's not yet present

Reply:
Maybe add a small comment to it, so this method won't confuse devs.